src/badguy/walking_badguy.hpp

```diff
@@ -24,6 +24,33 @@ class Timer;
 /** Base class for Badguys that walk on the floor. */
 class WalkingBadguy : public BadGuy
 {
+public:
+  enum class LedgeBehavior
+  {
+    /**
+     * Do not fall off any ledge at all.
+     */
+    STRICT,
+
+    /**
+     * Do not fall off any ledge, but still
+     * go down slopes.
+     */
+    SMART,
+
+    /**
+     * Fall off any ledge, unless the ledge
+     * is too tall (600px) or the ledge falls
+     * offscreen.
+     */
+    NORMAL,
+
+    /**
+     * Fall off any ledge.
+     */
+    FALL
+  };
+
```

Comment:
You could probably fit these comments on one line, and by removing the empty lines between the enumerators, this would take up way less lines.

Reply:
I think the names are fine. Kind of the same level of verbosity as HitResponse